src/dune/__main__.py

```diff
@@ -267,8 +272,148 @@ def load_all_modules_from_dir(plugin_dir):
+                elif len(args.update_app) >= 3:
+                    dune_sys.update_app(args.update_app[0],  # path
+                                        args.update_app[1],  # app name
+                                        args.update_app[2])  # lang
+
+            elif args.update_lib:
+                if len(args.update_lib) >= 5:
+                    dune_sys.update_lib(args.update_lib[0],  # path
+                                        args.update_lib[1],  # lib name
+                                        args.update_lib[2],  # lang
+                                        args.update_lib[3],  # compiler opts
+                                        args.update_lib[4])  # linker opts
+                elif len(args.update_lib) >= 4:
+                    dune_sys.update_lib(args.update_lib[0],  # path
+                                        args.update_lib[1],  # lib name
+                                        args.update_lib[2],  # lang
+                                        args.update_lib[3])  # compiler opts
+                elif len(args.update_lib) >= 3:
+                    dune_sys.update_lib(args.update_lib[0],  # path
+                                        args.update_lib[1],  # lib name
+                                        args.update_lib[2])  # lang
+            elif args.update_dep:
+                if len(args.update_dep) >= 6:
+                    dune_sys.update_dep(args.update_dep[0],  # project path
+                                        args.update_dep[1],  # object name (app/lib)
+                                        args.update_dep[2],  # dependency name
+                                        args.update_dep[3],  # location of dep
+                                        args.update_dep[4],  # tag/release number
+                                        args.update_dep[5])  # hash
+                if len(args.update_dep) >= 5:
+                    dune_sys.update_dep(args.update_dep[0],  # project path
+                                        args.update_dep[1],  # object name (app/lib)
+                                        args.update_dep[2],  # dependency name
+                                        args.update_dep[3],  # location of dep
+                                        args.update_dep[4])  # tag/release number
+                if len(args.update_dep) >= 4:
+                    dune_sys.update_dep(args.update_dep[0],  # project path
+                                        args.update_dep[1],  # object name (app/lib)
+                                        args.update_dep[2],  # dependency name
+                                        args.update_dep[3])  # location of dep
+                if len(args.update_dep) >= 3:
+                    dune_sys.update_dep(args.update_dep[0],  # project path
+                                        args.update_dep[1],  # object name (app/lib)
+                                        args.update_dep[2])  # dependency name
```

Comment:
> Same comment as add_args above. Please simplify.

I understand that it looks ugly but it happens because of the chaotic syntax of the whole CLI interface of DUNE. It make sense to redesign it, make it less ambiguous, use hierarchical approach with commands and their options, define a subparser for each command and describe the arguments correctly. In this case the parser will be able to generate correct help, recognize optional and demanded arguments correctly and we will not need to parse the arguments manually and add to their description "(Optional)". Parser should understand it itself. Anyway it is impossible to implement parsing of many optional arguments without a creation of a subparser object or manual parsing of the arguments. So I prefer to do it in a harmonious well designed system instead of adding crutches into the existing system. I decided current approach is the less painful temporary solution.